pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,522 @@
+			// 	amount,
+			// 	false,
+			// )?;
+
+			// Send the message through the router
+			Self::do_send_message(
+				who,
+				Message::Transfer {
+					pool_id,
+					tranche_id,
+					amount,
+					domain: address.clone().domain,
+					destination: address.clone().address,
+				},
+				address.domain,
+			)?;
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub(crate) fn now() -> Moment {
+			T::Time::now().as_secs()
+		}
+
+		pub fn do_set_domain_router(domain: Domain, router: Router<CurrencyIdOf<T>>) {
+			<DomainRouter<T>>::insert(domain.clone(), router.clone());
+			Self::deposit_event(Event::SetDomainRouter { domain, router });
+		}
+
+		pub fn do_send_message(
+			fee_payer: T::AccountId,
+			message: MessageOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let Router::Xcm(xcm_domain) =
+				<DomainRouter<T>>::get(domain.clone()).ok_or(Error::<T>::MissingRouter)?;
+
+			let contract_call = Self::encoded_contract_call(message.encode());
+			let ethereum_xcm_call =
+				Self::encoded_ethereum_xcm_call(xcm_domain.clone(), contract_call)
+					.map_err(|_| Error::<T>::FailedToBuildEthereumXcmCall)?;
+
+			pallet_xcm_transactor::Pallet::<T>::transact_through_sovereign(
+				T::Origin::root(),
+				// destination to which the message should be sent
+				Box::new(xcm_domain.location),
+				fee_payer,
+				// fee currency to be used
+				CurrencyPayment {
+					currency: Currency::AsCurrencyId(xcm_domain.fee_currency),
+					fee_amount: None,
+				},
+				// call to be executed in destination
+				ethereum_xcm_call,
+				// weight information to be used
+				OriginKind::SovereignAccount,
+				TransactWeights {
+					transact_required_weight_at_most: 8_000_000_000,
+					overall_weight: None,
+				},
+			)?;
+
+			Self::deposit_event(Event::MessageSent { message, domain });
+
+			Ok(())
+		}
+
+		/// Build the encoded `ethereum_xcm::transact(eth_tx)` call that should request to execute `contract_call`.
+		///
+		/// * `xcm_domain` - All the necessary info regarding the domain
+		/// where this `ethereum_xcm` call is to be executed
+		/// * `contract_call` - The encoded EVM call calling ConnectorsXcmRouter::handle(msg)
+		pub fn encoded_ethereum_xcm_call(
+			xcm_domain: XcmDomain<CurrencyIdOf<T>>,
+			evm_call: Vec<u8>,
+		) -> Result<Vec<u8>, ()> {
+			let eth_tx = xcm_primitives::EthereumXcmTransaction::V1(
+				xcm_primitives::EthereumXcmTransactionV1 {
+					gas_limit: U256::from(80_000),
+					fee_payment: xcm_primitives::EthereumXcmFee::Auto,
+					action: pallet_ethereum::TransactionAction::Call(xcm_domain.contract_address),
+					value: U256::zero(),
+					input: evm_call,
+					access_list: None,
+				},
+			);
+
+			// The complete encoded call payload to be sent in a `Transact` XCM instruction,
+			// encoding an ethereum_xcm::transact call.
+			let mut encoded: Vec<u8> = Vec::new();
+			encoded.append(&mut xcm_domain.ethereum_xcm_transact_call_index.clone());
+			encoded.append(&mut eth_tx.encode());
+
+			Ok(encoded)
```

Comment:
Change the layout to be more declarative:

1. append(call_index)
2. append(eth_tx)
3. return